cli/main.py

```diff
@@ -559,111 +762,239 @@ def simulate(config, action, nodes, federated_only, geth):
+
+            MinerEscrowDeployer._contract_name: DeployerInfo(deployer_class=MinerEscrowDeployer,
+                                                             upgradeable=True,
+                                                             agent_name='miner_agent',
+                                                             dependant='token_agent'),
+
+            PolicyManagerDeployer._contract_name: DeployerInfo(deployer_class=PolicyManagerDeployer,
+                                                               upgradeable=True,
+                                                               agent_name='policy_agent',
+                                                               dependant='miner_agent'
+                                                               ),
+
+            # UserEscrowDeployer._contract_name: DeployerInfo(deployer_class=UserEscrowDeployer,
+            #                                                 upgradeable=True,
+            #                                                 agent_name='user_agent',
+            #                                                 dependant='policy_agent'),  # TODO
+        })
+
+        click.confirm("Continue?", abort=True)
+        return deployers
+
+    if action == "contracts":
+        deployers = __get_deployers()
+        __deployment_transactions = dict()
+        __deployment_agents = dict()
+
+        available_deployers = ", ".join(deployers)
+        click.echo("\n-----------------------------------------------")
+        click.echo("Available Deployers: {}".format(available_deployers))
+        click.echo("Blockchain Provider URI ... {}".format(config.blockchain.interface.provider_uri))
+        click.echo("Registry Output Filepath .. {}".format(config.blockchain.interface.registry.filepath))
+        click.echo("Deployer's Address ........ {}".format(config.blockchain.interface.deployer_address))
+        click.echo("-----------------------------------------------\n")
+
+        def __deploy_contract(deployer_class: ClassVar,
+                              upgradeable: bool,
+                              agent_name: str,
+                              dependant: str = None
+                              ) -> Tuple[dict, EthereumContractAgent]:
+
+            __contract_name = deployer_class._contract_name
+
+            __deployer_init_args = dict(blockchain=config.blockchain,
+                                        deployer_address=config.blockchain.interface.deployer_address)
+
+            if dependant is not None:
+                __deployer_init_args.update({dependant: __deployment_agents[dependant]})
+
+            if upgradeable:
+                def __collect_secret_hash():
+                    secret = click.prompt("Enter secret hash for {}".format(__contract_name), hide_input=True, confirmation_prompt=True)
+                    secret_hash = hashlib.sha256(secret)
+                    if len(secret_hash) != 32:
```

Comment:
Why didn't this ever fail?
The previous line should probably be `hashlib.sha256(secret).digest()`.
Also, the `len(secret_hash)` will certainly always be `32`. What did you mean here?

Reply:
Ah, I thought you were asking about the reason for it.

Yeah, this is a bit wonky.

Reply:
I was using this code to debug while manually typing in the secret - Ill remove it and fix the hashing digest.  Thanks!

Reply:
Changed upstream in #470